### PR TITLE
Add option to make table headers sticky

### PR DESCRIPTION
**Pull request by rbeene**

### CODE REVIEW

File: addon/templates/components/table-columns.hbs [hunk 1]
```diff
@@ -5,7 +5,7 @@
         {{component column.headerComponent
           table=table
           column=column
-          classNames=column.headerClassNames
+          classNames=headerClassNames
```
Comment (cball) [inline addon/templates/components/table-columns.hbs]: we need this to keep the ability to set `headerClassNames` from the columns

File: addon/components/justa-table.js [hunk 2]
```diff
@@ -22,6 +22,20 @@ export default Component.extend({
     }
   },
 
+  didInsertElement() {
+    this._super();
```
Comment (cball) [inline addon/components/justa-table.js]: `this._super(...arguments)`

File: addon/components/justa-table.js [hunk 3]
```diff
@@ -22,6 +22,20 @@ export default Component.extend({
...
+    Ember.run.scheduleOnce('afterRender', this, this.afterRenderEvent);
```
Comment (cball) [inline addon/components/justa-table.js]: Seems like only doing this in `didInsertElement` will give problems when columns are added/removed. We should probably use `didRender` (https://guides.emberjs.com/v2.3.0/components/the-component-lifecycle/#toc_making-updates-to-the-rendered-dom-with-code-didrender-code) instead. something like this:

``` js
{
  didRender() {
    this._super(...arguments);
    this._uninstallStickyHeaders();
    this._installStickyHeaders();
  },

  willDestroyElement() {
    this._uninstallStickyHeaders()
  },

  _installStickyHeaders() {
    if (this.get('stickyHeader')) {
      this.$('table').stickyTableHeaders();
    }
  },

  _uninstallStickyHeaders() {
     if (this.get('stickyHeader')) {
      this.$('table').stickyTableHeaders('destroy');
    }
  }
}
```

Also note the added destroy call and using `this.$` to scope to the component's element.

File: .jshintrc [hunk 4]
```diff
@@ -13,6 +13,7 @@
   "evil": true,
   "forin": false,
   "immed": false,
+  "jquery": true,
```
Comment (cball) [inline .jshintrc]: shouldn't need this

File: addon/components/basic-header.js [hunk 5]
```diff
@@ -7,6 +7,7 @@ export default Ember.Component.extend({
   layout,
   tagName: 'th',
   classNameBindings: ['alignCenter:center', 'alignRight:right', 'textWrap'],
+  classNames: ['header-color'],
```
Comment (cball) [inline addon/components/basic-header.js]: Can we use `basic-header` instead to keep generic?

**Comment by cball** [comment]
Can you confirm:
1) adding/removing columns works ok with this? (should work ok with my comment above)
2) resizing columns works ok with this?

If there are issues with resize, you might need to call `trigger('resize.stickyTableHeaders');` here: https://github.com/cball/justa-table/blob/master/addon/components/table-columns.js#L156

File: addon/components/justa-table.js [hunk 6]
```diff
@@ -22,6 +22,34 @@ export default Component.extend({
     }
   },
 
+  didInsertElement() {
+    this._super(...arguments);
+    this._uninstallStickyHeaders();
+    Ember.run.scheduleOnce('afterRender', this, this._installStickyHeaders);
```
Comment (rbeene) [inline addon/components/justa-table.js:28]: Without using scheduleOnce, there were display issues with the columns changing in size when scrolling below the header.

**Comment by rbeene** [comment]
Resizing columns is currently NOT respected in the stickyHeader. However, adding new columns are properly added.

**Comment by rbeene** [comment]
Thoughts @cball?

File: addon/components/justa-table.js (hunk 6 above)
Comment (cball) [inline addon/components/justa-table.js:28]: right, thats why I said to use `didRender` vs `didInsertElement`

**Comment by cball** [comment]
closing in favor of a mostly css based approach.

**Pull request closed without merging**